PopPUNK/utils.py

```diff
@@ -161,6 +161,44 @@ def iterDistRows(refSeqs, querySeqs, self=True):
             for ref in refSeqs:
                 yield(ref, query)
 
+def listDistInts(refSeqs, querySeqs, self=True):
+    """Gets the ref and query ID for each row of the distance matrix
+
+    Returns an iterable with ref and query ID pairs by row.
+
+    Args:
+        refSeqs (list)
+            List of reference sequence names.
+        querySeqs (list)
+            List of query sequence names.
+        self (bool)
+            Whether a self-comparison, used when constructing a database.
+            Requires refSeqs == querySeqs
+            Default is True
+    Returns:
+        ref, query (str, str)
+            Iterable of tuples with ref and query names for each distMat row.
+    """
+    n = 0
+    num_ref = len(refSeqs)
+    num_query = len(querySeqs)
+    if self:
+        comparisons = [(0,0)] * int((num_ref * (num_ref-1)) * 0.5)
+        if refSeqs != querySeqs:
+            raise RuntimeError('refSeqs must equal querySeqs for db building (self = true)')
+        for i in range(num_ref):
+            for j in range(i + 1, num_ref):
+                comparisons[n] = (j, i)
+                n = n + 1
+    else:
+        comparisons = [(0,0)] * (len(refSeqs) * len(querySeqs))
+        for i in range(num_query):
+            for j in range(num_ref):
+                comparisons[n] = (j, i)
+                n = n + 1
+                
+    return comparisons
```

Comment:
```suggestion
    num_ref = len(refSeqs)
    num_query = len(querySeqs)
    if self:
        comparisons = [(0,0)] * int((num_ref * (num_ref-1)) * 0.5)
        if refSeqs != querySeqs:
            raise RuntimeError('refSeqs must equal querySeqs for db building (self = true)')
        for i in range(num_ref):
            for j in range(i + 1, num_ref):
                comparisons.append((j, i))
    else:
        comparisons = [(0,0)] * (len(refSeqs) * len(querySeqs))
        for i in range(num_query):
            for j in range(num_ref):
                comparisons.append((j, i))
                
    return comparisons
```

Reply:
Also, as per comment in #84, could a `yield()` approach be preferred here? Then the single function calls would become list comprehensions, and the function could be used directly as the iterable

Reply:
Will the latter option not append the actual pairings onto the zero-filled tuples, rather than replacing them? I was trying to preallocate the object with the correct size, not sure how much of an efficiency that is. Can try a `yield()` alternative.

Reply:
Yes, you're right that my change above needs to remove the preallocation to work. I think in python preallocation is less important than e.g. R, as `list.append()` seems to be pretty efficient, and is probably more pythonic, as they say. 
In any case, the `yield()` approach is likely to be neater.

Reply:
Converted to generator function in ecbdbe9.